cleanlab/internal/validation.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+Checks to ensure valid inputs for various methods.
+"""
+
+import warnings
+import numpy as np
+import pandas as pd
+from sklearn.utils import check_X_y
+
+
+def assert_valid_inputs(X, y, pred_probs=None):
+    """Checks that X, labels, and pred_probs are correctly formatted"""
+    if not isinstance(y, (list, np.ndarray, np.generic, pd.Series, pd.DataFrame)):
```

Comment:
I'm pretty sure (?) if this is passes, the rest of the new code you added below isnt necessary because the either shape or len will necessarily be defined?

Also if some reason you do need the code below, wouldn't a simpler logic just be

```python
try:
     len(X)
except:
    try:
         np.shape(X)
    except:
        raise TypeError("Features object X must support either: len(X) or X.shape[0]")
```

Reply:
This first line of code is only checking the labels `y`. The rest of code is checking the features `X` and `pred_probs`, as well as that the content of `y` is proper